[SVG2] Remove non-standard `kerning` property

Ahmad-S792 opened the pull request:
#### da471318105a38375ec838d0edb6d6787b9c4432
<pre>
[SVG2] Remove non-standard `kerning` property

<a href="https://bugs.webkit.org/show_bug.cgi?id=262847">https://bugs.webkit.org/show_bug.cgi?id=262847</a>
<a href="https://rdar.apple.com/116965514">rdar://116965514</a>

Reviewed by Simon Fraser.

This patch is to align WebKit with Gecko / Firefox, Blink / Chromium and
Web Specification [1].

[1] <a href="https://svgwg.org/svg2-draft/single-page.html#text-KerningProperty">https://svgwg.org/svg2-draft/single-page.html#text-KerningProperty</a>

It was discussed in W3C mailing list [2] to opt for CSS `font-kerning`.

[2] <a href="https://lists.w3.org/Archives/Public/www-svg/2012May/0101.html">https://lists.w3.org/Archives/Public/www-svg/2012May/0101.html</a>

It was never supported by Gecko / Firefox, while removed by Blnk in 2014 [3].

[3] <a href="https://github.com/chromium/chromium/commit/5f83d02185deb573a236459e03e069445808a981">https://github.com/chromium/chromium/commit/5f83d02185deb573a236459e03e069445808a981</a>

* Source/WebCore/animation/CSSPropertyAnimation.cpp:
(WebCore::CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap):
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::ComputedStyleExtractor::valueForPropertyInStyle const):
* Source/WebCore/css/SVGCSSComputedStyleDeclaration.cpp:
(WebCore::ComputedStyleExtractor::svgPropertyValue const):
* Source/WebCore/rendering/style/SVGRenderStyle.cpp:
(WebCore::SVGRenderStyle::SVGRenderStyle):
(WebCore::SVGRenderStyle::operator== const):
(WebCore::SVGRenderStyle::inheritedEqual const):
(WebCore::SVGRenderStyle::inheritFrom):
(WebCore::SVGRenderStyle::changeRequiresLayout const):
(WebCore::SVGRenderStyle::conservativelyCollectChangedAnimatableProperties const):
* Source/WebCore/rendering/style/SVGRenderStyle.h:
(WebCore::SVGRenderStyle::initialBaselineShiftValue):
(WebCore::SVGRenderStyle::strokeDashOffset const):
(WebCore::RenderStyle::hasVisibleStroke const):
(WebCore::RenderStyle::setFloodOpacity):
(WebCore::SVGRenderStyle::initialKerning): Deleted.
(WebCore::SVGRenderStyle::kerning const): Deleted.
(WebCore::RenderStyle::kerning const): Deleted.
(WebCore::RenderStyle::setKerning): Deleted.
(WebCore::SVGRenderStyle::setKerning): Deleted.
* Source/WebCore/rendering/style/SVGRenderStyleDefs.cpp:
(WebCore::StyleTextData::StyleTextData): Deleted.
(WebCore::StyleTextData::copy const): Deleted.
(WebCore::StyleTextData::operator== const): Deleted.
* Source/WebCore/rendering/style/SVGRenderStyleDefs.h:
(WebCore::StyleTextData::create): Deleted.
* Source/WebCore/rendering/svg/SVGTextLayoutEngine.cpp:
(WebCore::SVGTextLayoutEngine::layoutTextOnLineOrPath):
* Source/WebCore/rendering/svg/SVGTextLayoutEngineSpacing.cpp:
(WebCore::SVGTextLayoutEngineSpacing::calculateCSSSpacing):
(WebCore::SVGTextLayoutEngineSpacing::calculateCSSKerningAndSpacing): Deleted.
* Source/WebCore/rendering/svg/SVGTextLayoutEngineSpacing.h:
* Source/WebCore/svg/SVGElement.cpp:
(WebCore::SVGElement::cssPropertyIdForSVGAttributeName):
* Source/WebCore/svg/properties/SVGPropertyAnimatorFactory.h:
(WebCore::SVGPropertyAnimatorFactory::attributeAnimatorCreator):
* Source/WebCore/svg/svgattrs.in:
* LayoutTests/fast/css/getComputedStyle/computed-style-expected.txt:
* LayoutTests/fast/css/getComputedStyle/computed-style-without-renderer-expected.txt:
* LayoutTests/fast/css/getComputedStyle/resources/property-names.js:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-revert-layer-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-presentation-attribute-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/glib/svg/batik/text/textLayout-expected.txt:
* LayoutTests/platform/glib/svg/batik/text/textOnPathSpaces-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/ios/svg/batik/text/textLayout-expected.txt:
* LayoutTests/platform/ios/svg/batik/text/textOnPathSpaces-expected.txt:
* LayoutTests/platform/ipad/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/css/css-cascade/all-prop-revert-layer-expected.txt:
* LayoutTests/platform/mac/svg/batik/text/textLayout-expected.txt:
* LayoutTests/platform/mac/svg/batik/text/textOnPathSpaces-expected.txt:
* LayoutTests/svg/css/getComputedStyle-basic-expected.txt:
* LayoutTests/svg/css/parse-calc-length-expected.txt:
* LayoutTests/svg/css/parse-calc-length.html:
* LayoutTests/transitions/svg-transitions-expected.txt:
* LayoutTests/transitions/svg-transitions.html:
* LayoutTests/platform/gtk/svg/batik/text/textLayout-expected.png: Deleted.
* LayoutTests/platform/gtk/svg/batik/text/textLayout-expected.txt:
* LayoutTests/platform/gtk/svg/batik/text/textOnPathSpaces-expected.png: Deleted.
* LayoutTests/platform/gtk/svg/batik/text/textOnPathSpaces-expected.txt:

Canonical link: <a href="https://commits.webkit.org/283718@main">https://commits.webkit.org/283718@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bd22b59ebe19d1ecfcedb57825f14827d23ac5f1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/67120 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/46499 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/19752 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/71155 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/18253 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/69238 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/54297 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/18046 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/53830 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/12289 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/70187 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/42771 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/58080 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/34367 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/39443 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/15474 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/16607 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/61408 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/15815 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/72858 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/11079 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/15170 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/61303 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/11112 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/58139 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/61381 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14865 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/9109 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/2715 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/42304 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/43381 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/44567 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/43122 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nikolaszimmermann reviewed (commented):
[12 years ago](https://lists.w3.org/Archives/Public/www-svg/2012May/0114.html), I had no problem with this change, and still don't have :-)

Ahmad-S792 committed:
[SVG2] Remove non-standard `kerning` property

https://bugs.webkit.org/show_bug.cgi?id=262847
rdar://116965514

Reviewed by Simon Fraser.

This patch is to align WebKit with Gecko / Firefox, Blink / Chromium and
Web Specification [1].

[1] https://svgwg.org/svg2-draft/single-page.html#text-KerningProperty

It was discussed in W3C mailing list [2] to opt for CSS `font-kerning`.

[2] https://lists.w3.org/Archives/Public/www-svg/2012May/0101.html

It was never supported by Gecko / Firefox, while removed by Blnk in 2014 [3].

[3] https://github.com/chromium/chromium/commit/5f83d02185deb573a236459e03e069445808a981

* Source/WebCore/animation/CSSPropertyAnimation.cpp:
(WebCore::CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap):
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::ComputedStyleExtractor::valueForPropertyInStyle const):
* Source/WebCore/css/SVGCSSComputedStyleDeclaration.cpp:
(WebCore::ComputedStyleExtractor::svgPropertyValue const):
* Source/WebCore/rendering/style/SVGRenderStyle.cpp:
(WebCore::SVGRenderStyle::SVGRenderStyle):
(WebCore::SVGRenderStyle::operator== const):
(WebCore::SVGRenderStyle::inheritedEqual const):
(WebCore::SVGRenderStyle::inheritFrom):
(WebCore::SVGRenderStyle::changeRequiresLayout const):
(WebCore::SVGRenderStyle::conservativelyCollectChangedAnimatableProperties const):
* Source/WebCore/rendering/style/SVGRenderStyle.h:
(WebCore::SVGRenderStyle::initialBaselineShiftValue):
(WebCore::SVGRenderStyle::strokeDashOffset const):
(WebCore::RenderStyle::hasVisibleStroke const):
(WebCore::RenderStyle::setFloodOpacity):
(WebCore::SVGRenderStyle::initialKerning): Deleted.
(WebCore::SVGRenderStyle::kerning const): Deleted.
(WebCore::RenderStyle::kerning const): Deleted.
(WebCore::RenderStyle::setKerning): Deleted.
(WebCore::SVGRenderStyle::setKerning): Deleted.
* Source/WebCore/rendering/style/SVGRenderStyleDefs.cpp:
(WebCore::StyleTextData::StyleTextData): Deleted.
(WebCore::StyleTextData::copy const): Deleted.
(WebCore::StyleTextData::operator== const): Deleted.
* Source/WebCore/rendering/style/SVGRenderStyleDefs.h:
(WebCore::StyleTextData::create): Deleted.
* Source/WebCore/rendering/svg/SVGTextLayoutEngine.cpp:
(WebCore::SVGTextLayoutEngine::layoutTextOnLineOrPath):
* Source/WebCore/rendering/svg/SVGTextLayoutEngineSpacing.cpp:
(WebCore::SVGTextLayoutEngineSpacing::calculateCSSSpacing):
(WebCore::SVGTextLayoutEngineSpacing::calculateCSSKerningAndSpacing): Deleted.
* Source/WebCore/rendering/svg/SVGTextLayoutEngineSpacing.h:
* Source/WebCore/svg/SVGElement.cpp:
(WebCore::SVGElement::cssPropertyIdForSVGAttributeName):
* Source/WebCore/svg/properties/SVGPropertyAnimatorFactory.h:
(WebCore::SVGPropertyAnimatorFactory::attributeAnimatorCreator):
* Source/WebCore/svg/svgattrs.in:
* LayoutTests/fast/css/getComputedStyle/computed-style-expected.txt:
* LayoutTests/fast/css/getComputedStyle/computed-style-without-renderer-expected.txt:
* LayoutTests/fast/css/getComputedStyle/resources/property-names.js:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-revert-layer-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-presentation-attribute-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/glib/svg/batik/text/textLayout-expected.txt:
* LayoutTests/platform/glib/svg/batik/text/textOnPathSpaces-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/ios/svg/batik/text/textLayout-expected.txt:
* LayoutTests/platform/ios/svg/batik/text/textOnPathSpaces-expected.txt:
* LayoutTests/platform/ipad/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/css/css-cascade/all-prop-revert-layer-expected.txt:
* LayoutTests/platform/mac/svg/batik/text/textLayout-expected.txt:
* LayoutTests/platform/mac/svg/batik/text/textOnPathSpaces-expected.txt:
* LayoutTests/svg/css/getComputedStyle-basic-expected.txt:
* LayoutTests/svg/css/parse-calc-length-expected.txt:
* LayoutTests/svg/css/parse-calc-length.html:
* LayoutTests/transitions/svg-transitions-expected.txt:
* LayoutTests/transitions/svg-transitions.html:
* LayoutTests/platform/gtk/svg/batik/text/textLayout-expected.png: Deleted.
* LayoutTests/platform/gtk/svg/batik/text/textLayout-expected.txt:
* LayoutTests/platform/gtk/svg/batik/text/textOnPathSpaces-expected.png: Deleted.
* LayoutTests/platform/gtk/svg/batik/text/textOnPathSpaces-expected.txt:

Canonical link: https://commits.webkit.org/283718@main
